tapis-project/tapis-ui
# task/TUI-314 -- allow undefined values for app/system defaults

## Overview:

Allow undefined values for exec system and exec system logical queue, so that app or system defaults can be used

## Related Github Issues:

- [TUI-314](https://github.com/tapis-project/tapis-ui/issues/314)

## Summary of Changes:

- Add formikPatch that allows setting values in to undefined in FormikSelect
- Allow undefined defaults for execSystemId and execSystemLogicalQueue, so that they will fall back to the app specified exec system
- Add utility functions for computing exec system and queue defaults

## Testing Steps:

1. Test with both bsf v1 and NoExecSystem apps
2. Try using undefined defaults and seeing if the jobs can be submitted with undefined values
3. Try switching back and forth between fork and batch jobs (if the job is switched to batch and the previously selected system has no queues, the value should return to being undefined)

## UI Photos:

![image](https://user-images.githubusercontent.com/17181582/168893041-89b114d0-3db7-41ab-90ce-61d497abc2be.png)

## Notes:

If approved, please merge down to https://github.com/tapis-project/tapis-ui/pull/315 and approve there as well. (This is a fork of that PR)
